packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -1921,10 +1921,39 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
   }
 
   updatePartition(partition: string, partitionColumn: Column): void {
+    if (partitionColumn.type === 'char' && partition === '') {
+      return;
+    }
+
     const { model } = this.props;
-    const partitionFilter = partitionColumn
-      .filter()
-      .eq(model.dh.FilterValue.ofString(partition));
+
+    let partitionFilterValue;
+    if (
+      partitionColumn.type === 'java.lang.String' ||
+      partitionColumn.type === 'char'
+    ) {
+      partitionFilterValue = model.dh.FilterValue.ofString(
+        model.displayString(
+          partition,
+          partitionColumn.type,
+          partitionColumn.name
+        )
+      );
+    } else if (
+      partitionColumn.type === 'java.lang.Boolean' ||
+      partitionColumn.type === 'boolean'
+    ) {
+      partitionFilterValue = model.dh.FilterValue.ofBoolean(
+        typeof partition === 'string'
+          ? TableUtils.makeBooleanValue(partition)
+          : partition
+      );
+    } else {
+      partitionFilterValue = model.dh.FilterValue.ofNumber(partition);
```

Comment:
Did use `TableUtils.makeQuickFilter` not work?

Reply:
It used to break the `char` and `boolean` cases but those problems have been resolved so I have changed back to using `TableUtils.makeQuickFilterFromComponent`